draft-jennings-perc-double-01.md

```diff
@@ -96,127 +101,176 @@ Terms:
+~~~~~
 
-| Field      | Type | Value length |
-|------------|------|--------------|
-| X          | 1    | 1            |
-| CC         | 2    | 1            |
-| M          | 3    | 1            |
-| PT         | 4    | 1            |
-| Seq Num    | 5    | 2            |
-| Timestamp  | 6    | 4            |
-| SSRC       | 7    | 4            |
-| Ext Len    | 8    | 2            |
+If the OHB is three octets in length, it contains the original X bit,
+PT field value, and RTP packet sequence number.  In this case, the OHB has
+this form:
 
-The only values implementation MAY use are the values for X, PT, SeqNum, and
-ExtLen. The others entries are defined purely as place holders.
+~~~~~
+ 0                   1                   2
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3
++---------------+-------------------------------+
+|X|     PT      |        Sequence Number        |
++---------------+-------------------------------+
+~~~~~
 
-Open Issue: We could make a efficient coding by packing the above values as bits
-in bit field and perhaps packing some of the single values into the same byte.
+If an MDD modifies an original RTP header value, the MDD MUST include the
+OHB extension to reflect the changed value(s).  If another MDD along the
+media path makes additional changes to the RTP header and the original
+value is not already present in the OHB, the MDD must extend the OHB by
+adding the changed value to the OHB.  So as to properly preserve original
+RTP header values, an MDD MUST NOT change a value already present in the
+OHB extension.
 
-# Operations
+# RTP Operations
 
 
 ## Encrypting a Packet
 
 To encrypt a packet, the endpoint encrypts the packet with
-the inner transform, may add an OHB, then applies the outer transform.
+the inner transform and then applies the outer transform.  The processes
+is as follows:
 
 * Form an RTP packet.  If there are any header extensions, they MUST use
   {{RFC5285}}.
 
-* Apply the transform to the RTP packet
+* Apply the inner cryptographic transform to the RTP packet
 
-* Optionally add an OHB header extension.  The endpoint MAY include any header
-  fields that are signaled to be modified by the MDD, to reduce processing
-  burden on the MDD. Open Issue: do we want the sending client to be able to add
-  an OHB?
+* Optionally add an OHB header extension.  The endpoint MAY include the OHB
+  containing copies of RTP header information it knows through some means
+  defined outside of this document that the MDD is likely to change.
+{Note: I don't like this step, because it doesn't really save
+any work for the MDD.  The MDD has to look for the field and get prepared
+to insert it or change it.  By the time it does that, there is really
+very little work left.  And I don't like it being mandatory, either, because
+the only logical mandatory requirement would be the 3-octet variety, thus
+making every packet 3 octets longer.  I prefer we just leave this to the MDD.}
```

Comment:
The argument for doing this would be bandwidth does not change and packet size for MTU does not change but I think  you are right. Lets drop the stuff about the source inserting it. It always can anyways and it will just look like their was a prior MDD but we don't need to explicitly call it out.